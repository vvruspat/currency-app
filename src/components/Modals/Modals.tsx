import React, { useCallback, useState } from "react";
import {
  SelectModal,
  SelectModalProps,
} from "../uikit/Select/SelectModal/SelectModal";

export const ModalsContext = React.createContext<{
  setModal: (modal: ModalsProps | null) => void;
}>({
  setModal: (modal: ModalsProps | null) => {},
});

export enum MODALS {
  SELECT_MODAL = "select-modal",
  ERROR_MODAL = "error-modal",
  EXCHANGE_SUCCESS_MODAL = "exchange-success-modal",
}

// TODO: render and start modals from the component where it need

type ModalsProps = {
  modal: MODALS.SELECT_MODAL;
  props: SelectModalProps;
};
// | { modal: MODALS.ERROR_MODAL; props: /*ErrorModalProps*/ {} }
// | {
//     modal: MODALS.EXCHANGE_SUCCESS_MODAL;
//     props: /*ExchangeSuccessModalProps*/ {};
//   };

type ModalsRootProps = {};

export const ModalsRoot: React.FC<ModalsRootProps> = ({ children }) => {
  const [currentModal, setCurrentModal] = useState<ModalsProps | null>(null);

  const setModal = useCallback((modal: ModalsProps | null) => {
    setCurrentModal(modal);
  }, []);

  return (
    <ModalsContext.Provider value={{ setModal }}>
      {children}
      {/* <ErrorModal
        {...currentModalProps}
        show={currentModal === MODALS.ERROR_MODAL}
      /> */}
      <SelectModal
        {...currentModal?.props}
        show={currentModal?.modal === MODALS.SELECT_MODAL}
      />
      {/* <ExchangeSuccessModal
        {...currentModalProps}
        show={currentModal === MODALS.EXCHANGE_SUCCESS_MODAL}
      /> */}
    </ModalsContext.Provider>
  );
};
