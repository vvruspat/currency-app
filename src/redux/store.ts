import { configureStore } from "@reduxjs/toolkit";

import accountsReducer, { AccountsReducer } from "./accounts/accountsSlice";
import exchangeReducer, { ExchangeReducer } from "./exchange/exchangeSlice";

export type GlobalState = {
  accounts: AccountsReducer;
  exchange: ExchangeReducer;
};

export default configureStore<GlobalState>({
  reducer: {
    accounts: accountsReducer,
    exchange: exchangeReducer,
  },
});
