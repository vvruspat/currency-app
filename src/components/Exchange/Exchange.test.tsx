import { render, screen } from "@testing-library/react";
import { ReactNode } from "react";
import { Provider, ProviderProps } from "react-redux";
import { Exchange } from ".";
import store from "../../redux/store";
import { ModalsRoot } from "../Modals/Modals";

const ReduxProvider = ({ children, store }: ProviderProps) => (
  <Provider store={store}>{children}</Provider>
);

test("renders ExchangePage", () => {
  const wrapper = ({ children }: { children: ReactNode }) => (
    <ReduxProvider store={store}>
      <ModalsRoot>{children}</ModalsRoot>
    </ReduxProvider>
  );

  render(<Exchange data-testid="test-exchange-page" />, { wrapper });
  const ExchangeElement = screen.getByTestId("test-exchange-page");

  expect(ExchangeElement).toBeInTheDocument();
  expect(ExchangeElement).toMatchSnapshot();
});
