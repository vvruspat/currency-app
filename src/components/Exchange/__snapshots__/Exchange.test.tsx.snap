// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ExchangePage 1`] = `
<div
  class="currency-exchange"
  data-testid="test-exchange-page"
>
  <div
    class="spinner-wrapper"
  >
    <div
      class="spinner"
    />
  </div>
</div>
`;
